### DISCUSSION TITLE
Failed to build third party libraries for PS5

### DISCUSSION THREAD

**Discussion by ThiuT** [aws/aws-sdk-cpp#3927] [General]
### Describe the bug

Hello!

I'm trying to compile Gamelift libraries for PS5, but I get compilation errors that I don't know how to fix.

I use a PS5 specific CMake Toolchain file included with the PS5 SDK. I'm new to CMake and I don't know much about how to tweak the configuration and make this work.

I'm compiling from tag 1.8.x as it's the version used in my Unreal Engine integration and I need a matching version for all platforms.

Thanks!

### Expected Behavior

With the adequate configuration, I should be able to compile the libraries and use them in my Unreal Engine integration.

### Current Behavior

Compilation fails, seeloutput log attached: [awslogs.txt](https://github.com/aws/aws-sdk-cpp/files/9665680/awslogs.txt)

The compilation fails right at the start when compiling AwsCommon, because of implicit function declaration being invalid in C99. I tried changing the C standard but I'm not sure how since I'm new to CMake.

### Reproduction Steps

- Have access to the PS5 SDK
- Clone repository and pull 1.8.x tag
- Run: %PathToPS5SDK%\Prospero\Tools\CMake\PS5CMake.bat ./.. -DBUILD_ONLY="gamelift"

### AWS CPP SDK version used

1.8.x

### Compiler and Version used

prospero-clang 13.0.1

### Operating System and version

Windows 10 19043.2006

**Comment by jmklix** [comment]
This isn't something that we currently support, but might at some point in the future. Please upvote the issue if this is something that you would like support for.

I did take a look at the log file and it looks like you are running into a problem with AwsCCommon, but I'm not sure what the fix would be without looking into it more.

**Comment by ThiuT** [comment]
Hello!

Well I hope a lot of PS5 developers show up and upvote this, because I'm still stuck! From what I understood so far the issue with compiling AwsCommon was that it has different sources for windows and posix. PS5 being a Unix like target, AwsCommon tries to use posix sources, but fails because I do not have the required includes/libraries on my Windows environment. 

Editing CMakeLists.txt to force using windows sources solve the first few errors, but other errors pop up. So I'm starting to think that I should try setting up a Linux workspace and use it to try and build the libraries. Is there anything specific I should know to build Gamelift Libraries from a Linux environment?

Best regards,

**Comment by ThiuT** [comment]
Hello, just wondering if there has been any update on a plan to support PS5 and offer compiled libraries

**Comment by kai-ion** [comment]
Converting this to a Discussion. The ask is an effort we intend to pursue over time rather than a single change with a near-term milestone, so a Discussion is a better home for gathering use cases and tracking the pieces.
